### PR TITLE
Allow negative `minValue` in `NumberField` and validate the configured range at hydration time.

### PR DESCRIPTION
**Pull request by michael-toporkov**

### CODE REVIEW

File: tests/Field/Definition/NumberFieldTest.php [hunk 1]
```diff
@@ -233,4 +240,113 @@ public function testManagementApiData (NumberField $field, array $expected) : vo
...
+
+		yield "negative value without configured range" => [
+			new NumberField("label"),
+			"-100.5",
+		];
+
+		yield "null is valid when not required" => [
+			new NumberField("label", minValue: -5, maxValue: 5),
+			null,
+		];
```
Comment (apfelbox) [inline tests/Field/Definition/NumberFieldTest.php]: We should also add a test where `min` and `max` are negative

Comment (michael-toporkov) [reply tests/Field/Definition/NumberFieldTest.php]: Good point. Done 🙂
